pandas/core/arrays/base.py

```diff
@@ -682,6 +696,33 @@ def searchsorted(self, value, side="left", sorter=None):
         arr = self.astype(object)
         return arr.searchsorted(value, side=side, sorter=sorter)
 
+    def equals(self, other: "ExtensionArray") -> bool:
+        """
+        Return if another array is equivalent to this array.
+
+        Parameters
+        ----------
+        other: ExtensionArray
+            Array to compare to this Array.
+
+        Returns
+        -------
+        boolean
+            Whether the arrays are equivalent.
+
+        """
+        if not type(self) == type(other):
+            return False
+        elif not len(self) == len(other):
+            return False
+        else:
+            equal_values = self == other
+            if isinstance(equal_values, ExtensionArray):
+                # boolean array with NA -> fill with False
+                equal_values = equal_values.fillna(False)
+            equal_na = self.isna() & other.isna()
+            return (equal_values | equal_na).all().item()
```

Comment:
is the `.item()` necessary?

Reply:
> is the .item() necessary?

It's to have a python bool, instead of a numpy bool, as result. 
I added a comment to the tests to make it explicit this is the reason we are asserting with `is True/False` (and later on we don't inadvertedly "clean" that up)

Reply:
thanks for clarifying.  i was recently reminded that `np.array(np.timedelta64(1234, "ns")).item()` gives an int instead of timedelta64, so im now cautious around `.item()`